agent/pkg/handler/linkerd_info.go

```diff
@@ -8,32 +8,47 @@ import (
 	pb "github.com/buoyantio/linkerd-buoyant/gen/bcloud"
 	log "github.com/sirupsen/logrus"
 	"google.golang.org/protobuf/encoding/prototext"
+	"gopkg.in/yaml.v2"
 )
 
 const (
-	linkerdInfoInterval = time.Minute
+	linkerdInfoInterval       = time.Minute
+	diagnosticCollectDuration = time.Minute * 5
```

Comment:
Good question and I think that kind of relates again with the overlap between this feature and the logs one

Reply:
I think we should wait for more guidance from the Linkerd team here. It may not be appropriate to change the log level at all, or maybe we only want to set `trace` for a few libraries, and leave the rest at debug or info? Maybe 1 minute's worth of logs will suffice?

Regarding this comment:

> perhaps giving the user a separate, "Enable trace logging" button, or, stream metrics and logs back immediately, but also continue streaming trace logs as they come in?

The ability to change the proxy log level from the UI is on the feature roadmap, and I anticipate that we'd do that via a separate message type on the `ManageAgent` stream. Likewise we'll also have the ability to view recent logs for any given proxy, but that won't reuse the log collection code that we're adding for diagnostics, since the use cases are substantially different (get n most recent logs without adjusting the log level, versus collecting as many logs as we can for diagnostics).

Reply:
Alright, lets wait for input from the Linkerd team